internal/internal_workflow_client.go

```diff
@@ -1658,9 +1654,111 @@ func (w *workflowClientInterceptor) ExecuteWorkflow(
 		dataConverter:    w.client.dataConverter,
 		failureConverter: w.client.failureConverter,
 		registry:         w.client.registry,
+		startRequest:     startRequest,
 	}, nil
 }
 
+func (w *workflowClientInterceptor) executeWorkflowWithOperation(
+	ctx context.Context,
+	startRequest *workflowservice.StartWorkflowExecutionRequest,
+	operation StartWorkflowOperation,
+) (*workflowservice.StartWorkflowExecutionResponse, error) {
+	startOp := &workflowservice.ExecuteMultiOperationRequest_Operation{
+		Operation: &workflowservice.ExecuteMultiOperationRequest_Operation_StartWorkflow{
+			StartWorkflow: startRequest,
+		},
+	}
+
+	var withStartOp *workflowservice.ExecuteMultiOperationRequest_Operation
+	switch t := operation.(type) {
+	case *UpdateWorkflowOperation:
+		updateReq, err := w.createUpdateWorkflowRequest(ctx, t.input)
+		if err != nil {
+			return nil, fmt.Errorf("%w: %w", errInvalidWorkflowOperation, err)
+		}
+
+		if updateReq.WorkflowExecution.WorkflowId == "" {
+			updateReq.WorkflowExecution.WorkflowId = startRequest.WorkflowId
+		}
```

Comment:
If the user set the Workflow ID manually and it doesn't match with the Start Workflow's, the server will return a validation error.